spec/commands/binstubs_spec.rb

```diff
@@ -187,6 +187,67 @@
           end
         end
       end
+
+      context "when working with multiple gemfiles" do
+        let(:bundle_binstub) { bundled_app("bin/bundle") }
+        let(:load_bundle_exe) { "load Gem.bin_path(\"bundler\", \"bundle\")" }
+        let(:output_bundle_gemfile) { "puts ENV[\"BUNDLE_GEMFILE\"]" }
+
+        before do
+          build_repo2 do
+            build_gem "print_hello", "1.0" do |s|
+              s.executables = "print_hello"
+              s.bindir = "exe"
+              s.write "exe/print_hello", <<-R
+                puts "Hello!"
+              R
+            end
+            build_gem "print_goodbye", "1.0" do |s|
+              s.executables = "print_goodbye"
+              s.bindir = "exe"
+              s.write "exe/print_goodbye", <<-R
+                puts "Goodbye."
+              R
+            end
+          end
+
+          bundle! "config --global path #{bundled_app}"
+          bundle! "config --global bin #{bundled_app("bin")}"
+
+          install_gemfile! <<-G
+            source "file://#{gem_repo2}"
+          G
+          bundle! "binstubs bundler"
+        end
+
+        it "uses the correct Gemfile" do
+          create_file "one/Gemfile", <<-G
+            source "file://#{gem_repo2}"
+            gem "print_hello"
+          G
+          create_file "two/Gemfile", <<-G
+            source "file://#{gem_repo2}"
+            gem "print_goodbye"
+          G
+          Dir.chdir("one") { bundle! "install" }
+          Dir.chdir("two") { bundle! "install" }
+
+          binstub = File.read(bundled_app("bin/bundle"))
+          # Replace load statement with BUNDLE_GEMFILE puts
+          binstub.gsub!(load_bundle_exe, output_bundle_gemfile)
```

Comment:
can we put these in the binstub for the gem, so that way we won't have to mess with the file we're supposed to be testing?

Reply:
🤔 I don't believe so... I tried a few different ways, but `ENV['BUNDLE_GEMFILE']` is `nil` outside of the bundler binstub. It is [temporary assigned](https://github.com/bundler/bundler/blob/cecdfdb5b2a76133b0a83093ff6d80d1ffd97b46/lib/bundler/templates/Executable.bundler#L71) within the scope of the binstub execution.